File: Completed-Algorithms/LeetCode_49_Group_Anagrams/anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let hashMap = {}
    for (let i = 0; i < strs.length; i++) {
        let sortedStr = strs[i].split('').sort().join('')
        if (!hashMap.hasOwnProperty(sortedStr)) hashMap[sortedStr] = [strs[i]]
        else hashMap[sortedStr].push(strs[i]) 
    }
    return Object.values(hashMap)
};

File: Completed-Algorithms/LeetCode_49_Group_Anagrams/anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dictionary = {}
        for str in strs:
            sortedStr = "".join(sorted(str))
            if sortedStr not in dictionary:
                dictionary[sortedStr] = [str]
            else:
                dictionary[sortedStr].append(str)
        return list(dictionary.values())
        
